### ISSUE TITLE
FailureHandlingKind.Undo is missing

### ISSUE THREAD

**Issue by perrinjerome** [openlawlibrary/pygls#95]
`FailureHandlingKind` is [specified](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#initialize
) as

```typescript
export namespace FailureHandlingKind {

	/**
	 * Applying the workspace change is simply aborted if one of the changes provided
	 * fails. All operations executed before the failing operation stay executed.
	 */
	export const Abort: FailureHandlingKind = 'abort';

	/**
	 * All operations are executed transactionally. That means they either all
	 * succeed or no changes at all are applied to the workspace.
	 */
	export const Transactional: FailureHandlingKind = 'transactional';


	/**
	 * If the workspace edit contains only textual file changes they are executed transactionally.
	 * If resource changes (create, rename or delete file) are part of the change the failure
	 * handling strategy is abort.
	 */
	export const TextOnlyTransactional: FailureHandlingKind = 'textOnlyTransactional';

	/**
	 * The client tries to undo the operations already executed. But there is no
	 * guarantee that this succeeds.
	 */
	export const Undo: FailureHandlingKind = 'undo';
}
```

but pygls definition is:
https://github.com/openlawlibrary/pygls/blob/df8950dac6539e387817d3f310c18465d8997204/pygls/types.py#L572-L576

I'm thinking of a fix like:

```patch
diff --git a/pygls/types.py b/pygls/types.py
index 48b79d1..0642b57 100644
--- a/pygls/types.py
+++ b/pygls/types.py
@@ -574,7 +574,7 @@ class FailureHandlingKind(str, enum.Enum):
     Abort = 'abort'
     Transactional = 'transactional'
     TextOnlyTransactional = 'textOnlyTransactional'
-    FailureHandlingKind = 'undo'
+    Undo = 'undo'
 
 ```

Let's address this after https://github.com/openlawlibrary/pygls/pull/92 to prevent conflict.

**Comment by danixeee** [comment]
Closed via #92.

**Comment by perrinjerome** [comment]
> Closed via #92.

This was not part of #92 . I just submitted https://github.com/openlawlibrary/pygls/pull/98 to fix this issue.
